### PR TITLE
fix(behavior_path_planner): remove waitForData() in constructor

### PR DESCRIPTION
**Pull request by TakaHoribe**
## Description

Remove `waitForData()` in the constructor since it disturbs other nodes in the same component to launch.
Add `isDataReady()` instead at the beginning of the timer callback process.

## Tests performed

- run psim and check if there is no degradation.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.
- [x] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by tkimura4** [comment]
I confirmed that the behavior nodes were registered before a route was published by this PR.

image

File: planning/behavior_path_planner/src/behavior_path_planner_node.cpp [hunk 1]
```diff
@@ -495,46 +493,36 @@ BehaviorTreeManagerParam BehaviorPathPlannerNode::getBehaviorTreeManagerParam()
   return p;
 }
 
-void BehaviorPathPlannerNode::waitForData()
+// wait until mandatory data is ready
+bool BehaviorPathPlannerNode::isDataReady()
 {
-  // wait until mandatory data is ready
-  while (!current_scenario_ && rclcpp::ok()) {
-    RCLCPP_INFO_SKIPFIRST_THROTTLE(get_logger(), *get_clock(), 5000, "waiting for scenario topic");
```
Comment (taikitanaka3) [inline planning/behavior_path_planner/src/behavior_path_planner_node.cpp:502]: @TakaHoribe 
I think these missing info is useful for scenario simulator like this PR
https://github.com/autowarefoundation/autoware.universe/commit/05f6a6bafd545859d09413322def4f94d756a888

**Review by tkimura4 — APPROVED** [review]
LGTM

File: planning/behavior_path_planner/src/behavior_path_planner_node.cpp (hunk 1 above)
Comment (TakaHoribe) [reply planning/behavior_path_planner/src/behavior_path_planner_node.cpp:502]: add info print

**Review by taikitanaka3 — APPROVED** [review]
LGTM

**Pull request merged**